src/Polly/Registry/PolicyRegistry.cs

```diff
@@ -26,22 +26,9 @@ public PolicyRegistry()
     /// </summary>
     /// <param name="registry">a dictionary containing keys and policies used for testing.</param>
     internal PolicyRegistry(IDictionary<string, IsPolicy> registry)
-    {
-#pragma warning disable S3236 // Remove this argument from the method call; it hides the caller information.
-        Debug.Assert(registry != null, "This constructor is for testing only, and should not be called with a null registry.");
-#pragma warning restore S3236 // Remove this argument from the method call; it hides the caller information.
-        _registry = registry;
-    }
-
-    private ConcurrentDictionary<string, IsPolicy> ThrowIfNotConcurrentImplementation()
```

Comment:
This is redundant if we just enforce that the test constructor creates a concurrent dictionary.